mule-extensions-api/src/main/java/org/mule/runtime/extension/api/model/parameter/ImmutableParameterModel.java

```diff
@@ -174,32 +178,35 @@ public ImmutableParameterModel(String name,
-   * @param role                    this parameter's purpose
-   * @param dslConfiguration        a model which describes the DSL semantics for this parameter
-   * @param displayModel            a model which contains directive about how the parameter is displayed in the UI
-   * @param layoutModel             a model which contains directives about the parameter's layout in the UI
-   * @param allowedStereotypeModels A {@link Set} with the stereotypes of the allowed values
-   * @param modelProperties         A {@link Set} of custom properties which extend this model
-   * @param deprecationModel        a {@link DeprecationModel} describing if the parameter is deprecated. A null value means it is
-   *                                not deprecated.
-   * @param semanticTerms           a {@link Set} of semantic terms which describe the parameter's meaning and effect
+   * @param name                     the parameter's name. Cannot be blank.
+   * @param description              the parameter's description
+   * @param type                     the parameter's {@link MetadataType}. Cannot be {@code null}
+   * @param hasDynamicType           if the given {@code type} is of dynamic kind and has to be discovered during design time
+   * @param required                 whether this parameter is required or not
+   * @param isConfigOverride         whether this parameter is a configuration override or not
+   * @param isComponentId            whether this parameter serves as a {@link ComponentModel} ID or not
+   * @param expressionSupport        the {@link ExpressionSupport} that applies to {@code this} {@link ParameterModel}
+   * @param defaultValue             this parameter's default value
+   * @param role                     this parameter's purpose
+   * @param dslConfiguration         a model which describes the DSL semantics for this parameter
+   * @param displayModel             a model which contains directive about how the parameter is displayed in the UI
+   * @param layoutModel              a model which contains directives about the parameter's layout in the UI
+   * @param valueProviderModel       a value provider model
+   * @param allowedStereotypeModels  A {@link Set} with the stereotypes of the allowed values
+   * @param modelProperties          A {@link Set} of custom properties which extend this model
+   * @param deprecationModel         a {@link DeprecationModel} describing if the parameter is deprecated. A null value means it
+   *                                 is not deprecated.
+   * @param semanticTerms            a {@link Set} of semantic terms which describe the parameter's meaning and effect
+   * @param fieldValueProviderModels {@link List} of field value provider models.
    * @throws IllegalArgumentException if {@code required} is {@code true} and {@code defaultValue} is not {@code null} at the same
    *                                  time
+   *
```

Comment:
Should we create a new constructor with the new value? or since this one is available @since 1.4.0 and 1.4.0 has not been released we can modify it?

Reply:
I guess it's ok to use this one, otherwise, we'll end up with a new connector with any new commit :p